src/components/header-search-input/index.tsx

```diff
@@ -1,16 +1,43 @@
+import DevPopover from 'components/dev-popover'
 import { IconSlashSquare16 } from '@hashicorp/flight-icons/svg-react/slash-square-16'
 import { IconSearch16 } from '@hashicorp/flight-icons/svg-react/search-16'
+import classNames from 'classnames'
 import s from './style.module.css'
 
-// TODO: make this functional (ref: https://app.asana.com/0/1201010428539925/1201247589988631/f)
-const HeaderSearchInput: React.FC = () => (
-  <div className={s.headerSearchInput}>
-    <div>
-      <IconSearch16 className={s.searchIcon} />
-      <p>Search</p>
-    </div>
-    <IconSlashSquare16 />
-  </div>
-)
+interface HeaderSearchInputProps {
+  theme?: 'light' | 'dark'
```

Comment:
I think we'll want to address theming concerns soon, but i feel confident that we'll need some kind of prop interface. This component could be extended in the future to consume theme Context, at which point this `theme` prop would be a way to override context rather than the only way to set the component's theme.

For context, one alternative might be a superglobal toggling of CSS property values, but this can wreak havoc if we have "mixed" theme elements, such as this "dark" themed header and search input in an otherwise "light" themed view.

Reply:
I think we only have designs for dark mode in header components, but having this prop in place and ready to go will make it easier to implement a light theme design when it comes! 👏🏻